### PR TITLE
Added tests for Dataset, datafile, category and metadataValue. Refactored datafile controller (response format was changed to common form) 

### PR DESCRIPTION
**Pull request by KhrapkoVasyl**

### CODE REVIEW

File: src/js/__tests__/routes/category.test.js [hunk 1]
```diff
@@ -0,0 +1,136 @@
+'use strict';
+
+const app = require('../../app');
+const db = require('../../db/db');
+const associate = require('../../db/associate');
+const Category = require('../../db/models/Category');
```
Comment (Bogdan-Zinovij) [inline src/js/__tests__/routes/category.test.js:6]: the imported model is not used

**Review by Bogdan-Zinovij — CHANGES_REQUESTED** [review]

File: src/js/__tests__/routes/dataFile.test.js [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+    const body = JSON.parse(response.body);
+    const item = body.data.item;
+
+    expect(response.statusCode).toBe(200);
+    expect(typeof body).toBe('object');
+    expect(body).toHaveProperty('status', 'success');
+    expect(body).toHaveProperty('data');
+    expect(typeof body.data).toBe('object');
+    expect(typeof item).toBe('object');
+    expect(item.id).toBe(dataFileMock.id);
```
Comment (artemmatiushenko1) [inline src/js/__tests__/routes/dataFile.test.js:95]: You can use toHaveProperty() to check both: if propery exists and if it has a correct value

File: src/js/__tests__/routes/dataFile.test.js [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+    expect(item.dataSet).toBe(dataFileMock.dataSet);
```
Comment (artemmatiushenko1) [inline src/js/__tests__/routes/dataFile.test.js:96]: You can use toHaveProperty() to check both: if property exists and if it has a correct value

File: src/js/__tests__/routes/dataSet.test.js [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+    const body = JSON.parse(response.body);
+    const item = body.data.item;
+
+    expect(response.statusCode).toBe(200);
+    expect(typeof body).toBe('object');
+    expect(body).toHaveProperty('status', 'success');
+    expect(body).toHaveProperty('data');
+    expect(typeof body.data).toBe('object');
+    expect(typeof item).toBe('object');
+    expect(item.id).toBe(dataSetMock.id);
```
Comment (artemmatiushenko1) [inline src/js/__tests__/routes/dataSet.test.js:77]: You can use toHaveProperty() to check both: if property exists and if it has a correct value

File: src/js/__tests__/routes/dataSet.test.js [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+    expect(item.category).toBe(dataSetMock.category);
```
Comment (artemmatiushenko1) [inline src/js/__tests__/routes/dataSet.test.js:78]: You can use toHaveProperty() to check both: if property exists and if it has a correct value

**Review by artemmatiushenko1 — CHANGES_REQUESTED** [review]

**Pull request merged**